engine/components-api/src/main/java/com/cloud/vm/snapshot/VMSnapshotManager.java

```diff
@@ -28,15 +28,36 @@
 
 public interface VMSnapshotManager extends VMSnapshotService, Manager {
 
-    static final ConfigKey<Integer> VMSnapshotExpireInterval = new ConfigKey<Integer>("Advanced", Integer.class, "vmsnapshot.expire.interval", "-1",
-            "VM Snapshot expire interval in hours", true, ConfigKey.Scope.Account);
+    Integer defaultVirtualMachineSnapshotsMax = 10;
+    Integer defaultVirtualMachineJobCheckInterval = 3000;
+    Integer defaultVirtualMachineSnapshotExpireInterval = -1;
+    Integer defaultVirtualMachineSnapshotCreateWait = 1800;
 
-    public static final int VMSNAPSHOTMAX = 10;
+    ConfigKey<Integer> virtualMachineSnapshotExpireInterval = new ConfigKey<Integer>("Advanced",
+            Integer.class,
+            "vmsnapshot.expire.interval",
+            defaultVirtualMachineSnapshotExpireInterval.toString(),
+            "VM Snapshot expire interval in hours",
+            false,
+            ConfigKey.Scope.Account);
+
+    ConfigKey<Integer> virtualMachineSnapshotMax = new ConfigKey<Integer>("Advanced",
+            Integer.class, "vmsnapshot.max",
+            defaultVirtualMachineSnapshotsMax.toString(),
+            "Maximum number of snapshots which user can create for a virtual machine",
+            false);
+
+    ConfigKey<Integer> virtualMachineSnapshotCreateWait = new ConfigKey<Integer>("Advanced",
+            Integer.class, "vmsnapshot.create.wait",
+            defaultVirtualMachineSnapshotCreateWait.toString(),
+            "Maximum time to wait for a VM snapshot creation in seconds",
+            false);
 
     /**
      * Delete all VM snapshots belonging to one VM
-     * @param id, VM id
-     * @param type,
+     *
+     * @param id   VM id
+     * @param type
```

Comment:
what does type mean?
Also, there is an excessive use of space in the "id" description